keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/ec2/ClusterDeployStrategy.kt

```diff
@@ -0,0 +1,48 @@
+package com.netflix.spinnaker.keel.api.ec2
+
+import com.fasterxml.jackson.annotation.JsonSubTypes
+import com.fasterxml.jackson.annotation.JsonSubTypes.Type
+import com.fasterxml.jackson.annotation.JsonTypeInfo
+import com.fasterxml.jackson.annotation.JsonTypeInfo.As
+import com.fasterxml.jackson.annotation.JsonTypeInfo.Id
+import com.fasterxml.jackson.annotation.JsonTypeName
+import java.time.Duration
+import java.time.Duration.ZERO
+
+@JsonTypeInfo(
+  use = Id.NAME,
+  include = As.PROPERTY,
+  property = "strategy"
+)
+@JsonSubTypes(
+  Type(RedBlack::class),
+  Type(Highlander::class)
+)
+sealed class ClusterDeployStrategy {
+  abstract fun toOrcaJobProperties(): Map<String, Any?>
+}
+
+@JsonTypeName("red-black")
+data class RedBlack(
+  val rollbackOnFailure: Boolean = true,
+  val resizePreviousToZero: Boolean = false,
+  val maxServerGroups: Int = 2,
+  val delayBeforeDisable: Duration = ZERO,
+  val delayBeforeScaleDown: Duration = ZERO
```

Comment:
I wonder if we should default to this something better, maybe 15-30mins?

Reply:
oh, one more... default MEME folks discovered (esp in large ASG deploys). waiting for all 100% of instances to be health may fail. Because of that, most edge teams specify `targetHealthyDeployPercentage = 90`

Reply:
I wanted to retain consistency with the existing defaults in Deck. We can certainly have that discussion but it's very easy to change the default later.